frontend/src/pages/SignupPage/SignupPage.js

```diff
@@ -4,44 +4,169 @@ import Button from 'react-bootstrap/Button';
 import {useState} from 'react';
 import DropdownButton from 'react-bootstrap/DropdownButton';
 import Dropdown from 'react-bootstrap/Dropdown';
-
+import { useAtom, atom} from "jotai";
+import {currUser} from '../../atoms.js';
+import ErrorBar from '@/components/ErrorBar/ErrorBar';
+import {useRouter} from 'next/router'
+// import {Navigate, useNavigate} from 'react-router-dom';
 
 
 function SignupPage() {
+
+    const [user, setUser] = useAtom(currUser);
```

Comment:
Make sure to implement the logic where upon a successful sign-up or log-in, the user atom is set properly.